src/api/Snapshot.ts

```diff
@@ -341,6 +343,54 @@ export class Snapshot extends API {
     const vp = await this.getLatestScores(space, [address])
     return Object.values(vp)[0]
   }
+
+  async getAddressesVotesByDate(addresses: string[]) {
+    const batchSize = 5000
+    const query = `
+      query ProposalVotes($space: String!, $addresses: [String]!, $first: Int!, $skip: Int!) {
+        votes (
+          where: { space: $space, voter_in: $addresses}
+          first: $first, skip: $skip
+          orderBy: "created",
+          orderDirection: desc
+        ) {
+          voter
+          created
+        }
+      }
+    `
+
+    let votes: VoteEvent[] = []
+
+    let hasNext = true
+    let skip = 0
+    try {
+      while (hasNext) {
```

Comment:
TODO: all this try/catch and while(hasNext) could go into a method 

```typescript
function inBatches(codeToExecute, batchSize, onError)
```

or similar...WDYT?

Reply:
It's a great idea, thanks for asking!